src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -178,18 +188,24 @@ public async ValueTask SetupAsync()
                 s_http2ConnectionPreface.AsSpan().CopyTo(_outgoingBuffer.AvailableSpan);
                 _outgoingBuffer.Commit(s_http2ConnectionPreface.Length);
 
-                // Send SETTINGS frame.  Disable push promise.
-                FrameHeader.WriteTo(_outgoingBuffer.AvailableSpan, FrameHeader.SettingLength, FrameType.Settings, FrameFlags.None, streamId: 0);
+                // Send SETTINGS frame.  Disable push promise & set initial window size.
+                FrameHeader.WriteTo(_outgoingBuffer.AvailableSpan, 2*FrameHeader.SettingLength, FrameType.Settings, FrameFlags.None, streamId: 0);
                 _outgoingBuffer.Commit(FrameHeader.Size);
                 BinaryPrimitives.WriteUInt16BigEndian(_outgoingBuffer.AvailableSpan, (ushort)SettingId.EnablePush);
                 _outgoingBuffer.Commit(2);
                 BinaryPrimitives.WriteUInt32BigEndian(_outgoingBuffer.AvailableSpan, 0);
                 _outgoingBuffer.Commit(4);
+                BinaryPrimitives.WriteUInt16BigEndian(_outgoingBuffer.AvailableSpan, (ushort)SettingId.InitialWindowSize);
+                _outgoingBuffer.Commit(2);
+                BinaryPrimitives.WriteUInt32BigEndian(_outgoingBuffer.AvailableSpan, (uint)InitialStreamWindowSize);
+                _outgoingBuffer.Commit(4);
 
                 // Send initial connection-level WINDOW_UPDATE
+                uint windowUpdateAmount = (uint)(ConnectionWindowSize - InitialStreamWindowSize);
```

Comment:
Why is this using InitialStreamWindowSize instead of DefaultInitialConnectionWindowSize?

Reply:
Good point, will change.